persistent/Database/Persist/Sql/Raw.hs

```diff
@@ -74,12 +90,18 @@ getStmt sql = do
     liftIO $ getStmtConn conn sql
 
 getStmtConn :: SqlBackend -> Text -> IO Statement
-getStmtConn conn sql = do
+getStmtConn = getStmtConn' False
+
+getStmtConn' :: Bool -> SqlBackend -> Text -> IO Statement
+getStmtConn' useCursorIfPossible conn sql = do
     smap <- liftIO $ readIORef $ connStmtMap conn
     case Map.lookup sql smap of
         Just stmt -> return stmt
         Nothing -> do
-            stmt' <- liftIO $ connPrepare conn sql
+            stmt' <- liftIO $
+                case connPrepareCursor conn of
+                    Just prepareCursor | useCursorIfPossible -> prepareCursor sql
+                    _ -> connPrepare conn sql
```

Comment:
so this is where we switch - if we have `connPrepareCursor = Just ...` and we're trying to stream, then we do this thing. otherwise, we do the old thing.